server/src/main/java/org/apache/cassandra/sidecar/tasks/CassandraClusterSchema.java

```diff
@@ -0,0 +1,273 @@
+                              cdcDatabaseAccessor.partitioner(),
+                              tableIdCache,
+                              cdcDatabaseAccessor::getTableId,
+                              instanceFetcher,
+                              cassandraBridgeFactory);
+    }
+
+    private static Set<CqlTable> buildCdcTables(@NotNull String fullSchema,
+                                                @NotNull CdcDatabaseAccessor cdcDatabaseAccessor,
+                                                @NotNull ConcurrentHashMap<TableIdentifier, UUID> tableIdCache,
+                                                @NotNull InstanceMetadataFetcher instanceFetcher,
+                                                @NotNull CassandraBridgeFactory cassandraBridgeFactory)
+    {
+        return buildCdcTables(fullSchema,
+                              cdcDatabaseAccessor.partitioner(),
+                              tableIdCache,
+                              cdcDatabaseAccessor::getTableId,
+                              instanceFetcher,
+                              cassandraBridgeFactory);
+    }
+
+    private static Set<CqlTable> buildCdcTables(@NotNull final String fullSchema,
+                                                @NotNull final Partitioner partitioner,
+                                                @NotNull final ConcurrentHashMap<TableIdentifier, UUID> tableIdCache,
+                                                @NotNull final Function<TableIdentifier, UUID> tableIdLoaderFunction,
+                                                @NotNull final InstanceMetadataFetcher instanceFetcher,
+                                                @NotNull final CassandraBridgeFactory cassandraBridgeFactory)
+    {
+        Map<TableIdentifier, String> createStmts = CdcUtil.extractCdcTables(fullSchema);
+        Map<String, Set<String>> udtsPerKeyspace = createStmts.keySet()
+                                                              .stream()
+                                                              .map(TableIdentifier::keyspace)
+                                                              .distinct() // remove duplicated keyspace strings
+                                                              .collect(Collectors.toMap(Function.identity(),
+                                                                                        keyspace -> CqlUtils.extractUdts(fullSchema, keyspace)));
+
+        Map<TableIdentifier, UUID> tableIds = createStmts.keySet()
+                                                         .stream()
+                                                         .collect(Collectors.toMap(Function.identity(),
+                                                                                   id -> tableIdCache.computeIfAbsent(id, tableIdLoaderFunction)));
+
+        NodeSettings nodeSettings = instanceFetcher.callOnFirstAvailableInstance(instance-> instance.delegate().nodeSettings());
```

Comment:
We already have the nodeSettings in the callsite for this method, let's just use that

Reply:
as a matter of fact, we have the bridge, we should just pass the bridge